packages/cli/src/generate.ts

```diff
@@ -777,20 +777,22 @@ export async function generate(
       isJavascriptOutput
         ? typeDoc('import("gqty").QueryFetcher') + 'const queryFetcher'
         : 'const queryFetcher : QueryFetcher'
-    } = async function (query, variables) {
+    } = async function (query, variables, fetchOptions) {
         // Modify "${endpoint}" if needed
         const response = await fetch("${endpoint}", {
           method: "POST",
-          headers: {
-            "Content-Type": "application/json",
-          },
           body: JSON.stringify({
             query,
             variables,
           }),
           mode: "cors",
+          ...fetchOptions,
+          headers: {
+            'Content-Type': 'application/json',
+            ...fetchOptions?.headers,
+          },
```

Comment:
There are folks like to lower case everything in js/ts when it is allowed, do you think it's a good idea to leave it all to the users when they bring their own headers?

```suggestion
          headers: fetchOptions?.headers ?? { 'Content-Type': 'application/json' },
```

Reply:
doing `headers: fetchOptions?.headers ?? { 'Content-Type': 'application/json' },` doesn't make any sense, in that case just doing:

```ts
method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      query,
      variables,
    }),
    headers: {
      'Content-Type': 'application/json',
    },
    ...fetchOptions,
  });
```

will do it, but then, if the resolved call has custom headers it will require the user to also always add "content-type": "application/json" or related, which is the whole point of doing this:

```ts
headers: {
            'Content-Type': 'application/json',
            ...fetchOptions?.headers,
}
```

to simplify the usage of the fetch options

Reply:
but I think I understand where this is coming from, I will just put the "content-type" on the example of putting manual headers

Reply:
I'd say it's good enough for the intended audience, GraphQL is meant to be json for all of my known use cases.

When the user tries to put their own `content-type` in lower cases, Chrome will try to concatenate the two headers with a comma, e.g. `Content-Type: application/json, text/html`.

I honestly don't know how web servers will behave this way.

Reply:
yeah, I get where the conflicts might happen when having different names for the same header, it's better and also cleaner code for the query fetcher, just makes it a little bit uglier but also more explicit on the resolved usage side

Reply:
I changed the query fetcher / resolved example with headers